### PR TITLE
Implement Tabbed_View

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/66015

This PR refactors the code used to generate EA tabs to create a reusable and tested tabbed view solution.

### CODE REVIEW

File: codeception.dist.yml [hunk 1]
```diff
@@ -38,14 +38,14 @@ modules:
...
+            adminEmail: admin@tec.tri.be
             title: 'Event Common Tests'
             phpBinary: php
             language: ''
             plugins:
-                - tribe-common/tribe-common.php
+                - the-events-calendar/the-events-calendar.php
             activatePlugins:
-                - tribe-common/tribe-common.php
+                - the-eventtribe-common/tribe-common.php
```
Comment (barryhughes) [inline codeception.dist.yml]: Typo?

File: src/Tribe/Tabbed_View.php [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+	 * @param string $active
+	 */
+	public function set_active( $active ) {
+		$this->active = $active;
+	}
+
+	/**
+	 * A method to sort tabs by priority in ascending order.
+	 *
+	 * @access private
```
Comment (barryhughes) [inline src/Tribe/Tabbed_View.php]: Do we need this (since the method signature tells us as much) and isn't it protected in any case?

**Comment by tr1b0t** [comment]
:bangbang: This Pull Request is against a Central ticket for "Next Release". It shouldn't be merged until the Central ticket has been changed to "Current Release" or "CR Maintenance Release".

File: src/Tribe/Tabbed_View.php [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+		$this->items[ $tab_slug ] = $tab;
+
+		// Return the tab
+		return $tab;
+	}
+
+	/**
+	 * Returns all the registered tabs.
+	 *
+	 * @return Tribe__Tabbed_View__Tab[]
```
Comment (barryhughes) [inline src/Tribe/Tabbed_View.php]: Wasn't actually aware that this was a valid way to express an array-of-a-certain-type in docblocks until now, nice!

File: src/admin-views/tabbed-view/tabbed-view.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * @var Tribe__Tabbed_View $view
+ */
+
+/** @var Tribe__Tabbed_View__Tab[] $tabs */
+$tabs = $view->get_visibles();
+?>
+
+<?php if ( count( $tabs ) > 1 ) : ?>
+    <div class="tabbed-view-wrap wrap">
+		<?php if ( $view->get_label() ) : ?>
```
Comment (barryhughes) [inline src/admin-views/tabbed-view/tabbed-view.php]: The formatting/indentation here seems slightly odd to me (ditto for the `foreach` that follows) - shouldn't the control statements be tabbed back a bit and the HTML inside the blocks be tabbed in?

File: tests/wpunit/Tribe/Tabbed_View/TabTest.php [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+	/**
+	 * @test
+	 * it should not be active if another tab is active
+	 */
+	public function it_should_not_be_active_if_another_tab_is_active() {
+		$tab = $this->prophesize( \Tribe__Tabbed_View__Tab::class );
+		$tab->get_slug()->willReturn( 'bar' );
+		$this->tabbed_view->get_active()->willReturn( $tab->reveal() );
+
```
Comment (barryhughes) [inline tests/wpunit/Tribe/Tabbed_View/TabTest.php]: Kill this extra line?

File: tests/wpunit/Tribe/Tabbed_View/TabTest.php [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	/**
+	 * @test
+	 * it should be active if this is active
+	 */
+	public function it_should_be_active_if_this_is_active() {
+		$tab = $this->prophesize( \Tribe__Tabbed_View__Tab::class );
+		$tab->get_slug()->willReturn( 'foo' );
+		$this->tabbed_view->get_active()->willReturn( $tab->reveal() );
+
```
Comment (barryhughes) [inline tests/wpunit/Tribe/Tabbed_View/TabTest.php]: Kill this extra line?

File: tests/wpunit/Tribe/Tabbed_View/TabTest.php [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	/**
+	 * @test
+	 * it should not be active if active is undefined on tabbed view
+	 */
+	public function it_should_not_be_active_if_active_is_undefined_on_tabbed_view() {
+		$tab = $this->prophesize( \Tribe__Tabbed_View__Tab::class );
+		$this->tabbed_view->get_active()->willReturn( false );
+
```
Comment (barryhughes) [inline tests/wpunit/Tribe/Tabbed_View/TabTest.php]: Kill this extra line?

**Comment by lucatume** [comment]
Updated

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Echoed strings should not be bracketed | [tabbed-view.php:21](https://github.com/moderntribe/tribe-common/blob/6c469efcd25302166866121eaf4899a2294b7986/src/admin-views/tabbed-view/tabbed-view.php#L21)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$tab' | [tabbed-view.php:21](https://github.com/moderntribe/tribe-common/blob/6c469efcd25302166866121eaf4899a2294b7986/src/admin-views/tabbed-view/tabbed-view.php#L21)

> via jenkins/codesniffer-phpcs

**Pull request merged**